src/main/java/de/bayern/gdi/gui/WMSMapSwing.java

```diff
@@ -159,53 +158,43 @@ public WMSMapSwing() {
      * @param mapURL The URL of the WMS Service
      * @throws MalformedURLException
      */
-    public WMSMapSwing(String mapURL, int width, int heigth) throws
+    public WMSMapSwing(String mapURL, int width, int heigth, String layer)
+            throws
             MalformedURLException {
-        this(new URL(mapURL), width, heigth);
+        this(new URL(mapURL), width, heigth, layer);
     }
 
     /**
      * Constructor.
      *
      * @param mapURL The URL of the WMS Service
      */
-    public WMSMapSwing(URL mapURL, int width, int heigth) {
+    public WMSMapSwing(URL mapURL, int width, int heigth, String layer) {
         initGeotoolsLocale();
         try {
             this.mapHeight = heigth;
             this.mapWidth = width;
             this.vBox = new VBox();
             this.wms = new WebMapServer(mapURL);
             List<Layer> layers = this.wms.getCapabilities().getLayerList();
-            ObservableList<String> layerList = FXCollections
-                    .observableArrayList();
-            for (Layer layer : layers) {
-                layerList.add(layer.getName());
-            }
-            if (layers == null) {
-                throw new ServiceException("could not connect to url");
+            Layer myLayer = null;
+            for (Layer wmsLayer: layers) {
+                if (wmsLayer.getTitle().toLowerCase().equals(
+                        layer.toLowerCase())) {
+                    myLayer = wmsLayer;
+                    break;
+                }
```

Comment:
Why not creating a layer by hand instead of getting, parsing, looping the whole capabilities document?